File: rs/earthbucks_lib/src/tx_signature.rs
pub struct TxSignature {
    pub hash_type: u8,
    pub sig_buf: [u8; 64],
}

impl TxSignature {
    pub const SIGHASH_ALL: u8 = 0x00000001;
    pub const SIGHASH_NONE: u8 = 0x00000002;
    pub const SIGHASH_SINGLE: u8 = 0x00000003;
    pub const SIGHASH_ANYONECANPAY: u8 = 0x00000080;
    pub const SIZE: usize = 65; // hashtype (1) plus r (32) plus s (32)

    pub fn new(hash_type: u8, sig_buf: [u8; 64]) -> Self {
        Self { hash_type, sig_buf }
    }

    pub fn to_buf(&self) -> [u8; TxSignature::SIZE] {
        let mut result = Vec::new();
        result.push(self.hash_type);
        result.extend(&self.sig_buf);
        result.try_into().unwrap()
    }

    pub fn from_buf(data: Vec<u8>) -> Result<Self, String> {
        if data.len() != TxSignature::SIZE {
            return Err("Invalid buffer length".to_string());
        }
        let hash_type = data[0];
        let sig_buf: [u8; 64] = data[1..65].try_into().unwrap();
        Ok(Self { hash_type, sig_buf })
    }
}

File: ts/npm-earthbucks-ebx-lib/src/tx-signature.ts
import { WebBuf } from "@webbuf/webbuf";
import { FixedBuf } from "@webbuf/fixedbuf";
import { U8, U16BE, U32BE, U64BE } from "@webbuf/numbers";

export class TxSignature {
  static readonly SIGHASH_ALL = new U8(0x00000001);
  static readonly SIGHASH_NONE = new U8(0x00000002);
  static readonly SIGHASH_SINGLE = new U8(0x00000003);
  static readonly SIGHASH_ANYONECANPAY = new U8(0x00000080);
  static readonly SIZE = 65;

  hashType: U8;
  sigBuf: FixedBuf<64>;

  constructor(hashType: U8, sigBuf: FixedBuf<64>) {
    this.hashType = hashType;
    this.sigBuf = sigBuf;
  }

  toBuf(): WebBuf {
    const hashTypeBuf = WebBuf.alloc(1);
    hashTypeBuf[0] = this.hashType.n;
    return WebBuf.concat([hashTypeBuf, this.sigBuf.buf]);
  }

  static fromBuf(buf: WebBuf): TxSignature {
    if (buf.length !== TxSignature.SIZE) {
      throw new Error("Invalid TxSignature length");
    }
    const hashType = new U8(buf[0] as number);
    const sigBuf = buf.subarray(1);
    const sigFixedEbxBuf = FixedBuf.fromBuf(64, sigBuf);
    return new TxSignature(hashType, sigFixedEbxBuf);
  }
}
